packages/contracts-bedrock/test/vendor/eas/EAS.t.sol

```diff
@@ -0,0 +1,3476 @@
+// SPDX-License-Identifier: MIT
+pragma solidity =0.8.15;
+
+// imports
+import { CommonTest } from "test/setup/CommonTest.sol";
+import { IEAS } from "src/vendor/eas/IEAS.sol";
+import { ISchemaRegistry } from "src/vendor/eas/ISchemaRegistry.sol";
+import { Attestation, AttestationRequest, AttestationRequestData, MultiAttestationRequest, RevocationRequest, DelegatedRevocationRequest, RevocationRequestData, MultiDelegatedAttestationRequest, MultiDelegatedRevocationRequest, DelegatedAttestationRequest, MultiRevocationRequest, Signature } from "src/vendor/eas/IEAS.sol";
+import { ISchemaResolver } from "src/vendor/eas/resolver/ISchemaResolver.sol";
+import { Predeploys } from "src/libraries/Predeploys.sol";
+import { EIP712 } from "@openzeppelin/contracts/utils/cryptography/draft-EIP712.sol";
+import { IERC1271 } from "@openzeppelin/contracts/interfaces/IERC1271.sol";
+import { ISemver } from "src/universal/interfaces/ISemver.sol";
+import { Strings } from "@openzeppelin/contracts/utils/Strings.sol";
+import { ECDSA } from "@openzeppelin/contracts/utils/cryptography/ECDSA.sol";
+
+// Mock contracts
+interface IEAS_EIP1271 {
+    function getNonce(address account) external view returns (uint256);
+}
+
+/// @dev Helper contract for testing EIP712 signature verification
+contract TestEIP712Helper is EIP712 {
+    constructor() EIP712("EAS", "1.3.0") {}
+
+    /// @dev Exposes internal hash function for testing
+    function hashTypedDataV4(bytes32 structHash) public view returns (bytes32) {
+        return _hashTypedDataV4(structHash);
+    }
+}
+
+/// @dev Proxy contract for testing EIP712 signature verification
+contract TestEIP712Proxy is EIP712 {
+    address public verifyingContract;
+    string private _name;
+
+    constructor(
+        address _verifyingContract,
+        string memory name_
+    ) EIP712(name_, "1.3.0") {
+        verifyingContract = _verifyingContract;
+        _name = name_;
+    }
+
+    /// @dev Returns the name of the contract
+    function name() public view returns (string memory) {
+        return _name;
+    }
+
+    /// @dev Returns the domain separator for EIP712
+    function DOMAIN_SEPARATOR() public view returns (bytes32) {
+        return _domainSeparatorV4();
+    }
+
+    /// @dev Exposes internal hash function for testing
+    function hashTypedDataV4(bytes32 structHash) public view returns (bytes32) {
+        return _hashTypedDataV4(structHash);
+    }
+
+    /// @dev Forwards attestation request to the verifying contract
+    function attestByDelegation(
+        DelegatedAttestationRequest calldata request
+    ) external returns (bytes32) {
+        return IEAS(verifyingContract).attestByDelegation(request);
+    }
+}
+
+// Mainnet Contract
+contract EASTest is CommonTest {
+    // =============================================================
+    //                           CONSTANTS
```

Comment:
You should delete these big spacers, will be more likely to be merged, none of the upstream code has these